<h1> Palmer Penguins </h1>
<hr>
<h3> Analysis: Castillano </h3>

In [1]:
import numpy as np

In [3]:
import pandas as pd

In [4]:
Data = pd.read_csv('Datasets\\penguins.csv')

In [5]:
Data

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1> EXPLORATORY DAAT ANALYTICS(EDA) </h1>

<h1>1. How many Male has Identify in year 2008?</h1>

In [9]:
y_2008 = Data[Data['year'] == 2008]

In [19]:
y_2008[y_2008['sex'] == 'male']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
51      Adelie     Biscoe            40.1           18.9              188.0   
53      Adelie     Biscoe            42.0           19.5              200.0   
55      Adelie     Biscoe            41.4           18.6              191.0   
57      Adelie     Biscoe            40.6           18.8              193.0   
59      Adelie     Biscoe            37.6           19.1              194.0   
61      Adelie     Biscoe            41.3           21.1              195.0   
63      Adelie     Biscoe            41.1           18.2              192.0   
65      Adelie     Biscoe            41.6           18.0              192.0   
67      Adelie     Biscoe            41.1           19.1              188.0   
69      Adelie  Torgersen            41.8           19.4              198.0   
71      Adelie  Torgersen            39.7           18.4              190.0   
73      Adelie  Torgersen            45.8           18.9              197.0   
75      Adelie  Torgersen            42.8           18.5              195.0   
77      Adelie  Torgersen            37.2           19.4              184.0   
79      Adelie  Torgersen            42.1           19.1              195.0   
81      Adelie  Torgersen            42.9           17.6              196.0   
83      Adelie  Torgersen            35.1           19.4              193.0   
85      Adelie      Dream            41.3           20.3              194.0   
86      Adelie      Dream            36.3           19.5              190.0   
88      Adelie      Dream            38.3           19.2              189.0   
91      Adelie      Dream            41.1           18.1              205.0   
93      Adelie      Dream            39.6           18.1              186.0   
95      Adelie      Dream            40.8           18.9              208.0   
97      Adelie      Dream            40.3           18.5              196.0   
99      Adelie      Dream            43.2           18.5              192.0   
187     Gentoo     Biscoe            48.4           16.3              220.0   
189     Gentoo     Biscoe            44.4           17.3              219.0   
191     Gentoo     Biscoe            48.7           15.7              208.0   
193     Gentoo     Biscoe            49.6           16.0              225.0   
195     Gentoo     Biscoe            49.6           15.0              216.0   
196     Gentoo     Biscoe            50.5           15.9              222.0   
199     Gentoo     Biscoe            50.5           15.9              225.0   
201     Gentoo     Biscoe            45.2           15.8              215.0   
203     Gentoo     Biscoe            48.5           14.1              220.0   
205     Gentoo     Biscoe            50.1           15.0              225.0   
207     Gentoo     Biscoe            45.0           15.4              220.0   
209     Gentoo     Biscoe            45.5           15.0              220.0   
211     Gentoo     Biscoe            50.4           15.3              224.0   
213     Gentoo     Biscoe            46.2           14.9              221.0   
215     Gentoo     Biscoe            54.3           15.7              231.0   
217     Gentoo     Biscoe            49.8           16.8              230.0   
219     Gentoo     Biscoe            49.5           16.2              229.0   
221     Gentoo     Biscoe            50.7           15.0              223.0   
223     Gentoo     Biscoe            46.4           15.6              221.0   
224     Gentoo     Biscoe            48.2           15.6              221.0   
227     Gentoo     Biscoe            48.6           16.0              230.0   
229     Gentoo     Biscoe            51.1           16.3              220.0   
231     Gentoo     Biscoe            45.2           16.4              223.0   
303  Chinstrap      Dream            49.5           19.0              200.0   
305  Chinstrap      Dream            52.8           20.0 

In [20]:
y_2008[y_2008['sex'] == 'male'].value_counts().sum()

np.int64(57)

<h1>INSIGHT</h1>
<hr>
<h3>The total Male Penguins that been identify in year 2008 is 57. </h3>

<hr>

<h1>2. How many Female has Identify in year 2007?</h1>

In [21]:
y_2007 = Data[Data['year'] == 2007]

In [23]:
y_2007[y_2007['sex'] == 'female']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
4       Adelie  Torgersen            36.7           19.3              193.0   
6       Adelie  Torgersen            38.9           17.8              181.0   
12      Adelie  Torgersen            41.1           17.6              182.0   
15      Adelie  Torgersen            36.6           17.8              185.0   
16      Adelie  Torgersen            38.7           19.0              195.0   
18      Adelie  Torgersen            34.4           18.4              184.0   
20      Adelie     Biscoe            37.8           18.3              174.0   
22      Adelie     Biscoe            35.9           19.2              189.0   
25      Adelie     Biscoe            35.3           18.9              187.0   
27      Adelie     Biscoe            40.5           17.9              187.0   
28      Adelie     Biscoe            37.9           18.6              172.0   
30      Adelie      Dream            39.5           16.7              178.0   
32      Adelie      Dream            39.5           17.8              188.0   
34      Adelie      Dream            36.4           17.0              195.0   
37      Adelie      Dream            42.2           18.5              180.0   
38      Adelie      Dream            37.6           19.3              181.0   
40      Adelie      Dream            36.5           18.0              182.0   
42      Adelie      Dream            36.0           18.5              186.0   
44      Adelie      Dream            37.0           16.9              185.0   
48      Adelie      Dream            36.0           17.9              190.0   
152     Gentoo     Biscoe            46.1           13.2              211.0   
154     Gentoo     Biscoe            48.7           14.1              210.0   
157     Gentoo     Biscoe            46.5           13.5              210.0   
158     Gentoo     Biscoe            45.4           14.6              211.0   
160     Gentoo     Biscoe            43.3           13.4              209.0   
162     Gentoo     Biscoe            40.9           13.7              214.0   
164     Gentoo     Biscoe            45.5           13.7              214.0   
166     Gentoo     Biscoe            45.8           14.6              210.0   
168     Gentoo     Biscoe            42.0           13.5              210.0   
170     Gentoo     Biscoe            46.2           14.5              209.0   
173     Gentoo     Biscoe            45.1           14.5              215.0   
174     Gentoo     Biscoe            46.5           14.5              213.0   
176     Gentoo     Biscoe            42.9           13.1              215.0   
180     Gentoo     Biscoe            48.2           14.3              210.0   
183     Gentoo     Biscoe            42.8           14.2              209.0   
184     Gentoo     Biscoe            45.1           14.5              207.0   
276  Chinstrap      Dream            46.5           17.9              192.0   
279  Chinstrap      Dream            45.4           18.7              188.0   
281  Chinstrap      Dream            45.2           17.8              198.0   
282  Chinstrap      Dream            46.1           18.2              178.0   
284  Chinstrap      Dream            46.0           18.9              195.0   
286  Chinstrap      Dream            46.6           17.8              193.0   
288  Chinstrap      Dream            47.0           17.3              185.0   
290  Chinstrap      Dream            45.9           17.1              190.0   
293  Chinstrap      Dream            58.0           17.8              181.0   
294  Chinstrap      Dream            46.4           18.6              190.0   
296  Chinstrap      Dream            42.4           17.3              181.0   
298  Chinstrap      Dream            43.2           16.6 

In [29]:
y_2007[y_2007['sex'] == 'female'].value_counts().sum()

np.int64(51)

<h1>INSIGHT</h1>
<hr>
<h3>The total Female Penguins in year 2007 is 51 Penguins.</h3>

<hr>

<h1>3. Which island is has a more Penguins?</h1>

In [30]:
Data['island'].unique()

array(['Torgersen', 'Biscoe', 'Dream'], dtype=object)

In [33]:
Data[Data['island'] == 'Torgersen'].value_counts().sum()

np.int64(47)

In [34]:
Data[Data['island'] == 'Biscoe'].value_counts().sum()

np.int64(163)

In [35]:
Data[Data['island'] == 'Dream'].value_counts().sum()

np.int64(123)

<h1>INSGHT</h1>
<hr>
<h3>Here's the summation is each type of island. the highest Population of Penguins is the island of 'Bescoe' total of  163 Penguins. </h3>

<hr>

<h1>4. Which Species has the lowest body_mass_g?</h1>

In [36]:
Data['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [38]:
adelie_df = Data[Data['species'] == 'Adelie']

In [39]:
adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

In [40]:
gentoo_df = Data[Data['species'] == 'Gentoo']

In [41]:
gentoo_df['body_mass_g'].mean()

np.float64(5076.016260162602)

In [42]:
chinstrap_df = Data[Data['species'] == 'Chinstrap']

In [43]:
chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

<h1>INSIGHT</h1>
<hr>
<h3>Base on the computation Adelie species has the lowest body_mass_g in all species, The body_mass_g is '3700.66... grams'. </h3>

<hr>

<h1>5. How much is the longest bill_length_mm in the three types penguins species?</h1>

In [44]:
Data['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

<h4>(for this question i will use the same DataFrame on question no.4)</h4>

In [46]:
adelie_df['bill_length_mm'].mean()

np.float64(38.79139072847682)

In [47]:
gentoo_df['bill_length_mm'].mean()

np.float64(47.50487804878048)

In [48]:
chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

<h1>INSIGHT</h1>
<hr>
<h3>In the three species Chinstrap Penguins has the longest bill_leng_mm Average, in total of '48.83...mm length'.</h3>

<hr>

<h1>6. Which Year has the Highest Counts of Penguins?</h1>

In [50]:
Data['year'].unique()

array([2007, 2008, 2009])

<h4>(for this question i've done to make the year Dataframe above, so i will make one more df in year 2009)</h4>

In [65]:
y_2007.value_counts().sum()

np.int64(103)

In [63]:
y_2008.value_counts().sum()

np.int64(113)

In [57]:
y_2009 = Data[Data['year'] == 2009]

In [64]:
y_2009.value_counts().sum()

np.int64(117)

<h1>INSIGHT</h1>
<hr>
<h3>The year has been have a highest counts of penguins is Year 2009, total of 117 Penguins.</h3>

<hr>

<h1>7. In Year 2007 to 2009, How many Male penguins that Adelie species got?</h1>

<h4>(for this question i will use the all DataFrame i created.)</h4>

In [71]:
year_2007 = adelie_df[adelie_df['year'] == 2007]

In [77]:
year_2007[year_2007['sex'] == 'male'].value_counts().sum()

np.int64(22)

In [78]:
year_2008 = adelie_df[adelie_df['year'] == 2008]

In [79]:
year_2008[year_2008['sex'] == 'male'].value_counts().sum()

np.int64(25)

In [80]:
year_2009 = adelie_df[adelie_df['year'] == 2009]

In [81]:
year_2009[year_2009['sex'] == 'male'].value_counts().sum()

np.int64(26)

In [85]:
adelie_df[adelie_df['sex'] == 'male'].value_counts().sum()

np.int64(73)

<h1>INSIGHT</h1>
<hr>
<h3>Here's the summation of each year that has a Adelie Male Penguins,</h3>
<h3>In Year 2007 the total of Adelie Male Penguins is 22,</h3>
<h3>In Year 2008 the total of Adelie Male Penguins is 25,</h3>
<h3>Lastly Year 2009 total a Adelie Male Peneguins is 26.</h3>
<h3>The total of Adelie Male Penguins is 73.</h3>